### PR TITLE
consolidate all the run commands

### PR DESCRIPTION
**Pull request by kellyselden**
lots of duplicate code has been eliminated. makes running a command cross platform easier/less likely to screw up.

Depends on #18, #19, #20

### CODE REVIEW

**Comment by kellyselden** [comment]
This is ready to go now. It also fixes Windows.

**Comment by kellyselden** [comment]
Windows support has arrived! Could use a code review though. Some core commands have changed a little.

File: lib/utilities/pristine.js [hunk 1]
```diff
@@ -85,7 +86,9 @@ function installPristineApp(appName, options) {
 
   chdir(temp.pristinePath);
 
-  var promise = applyCommand('new', appName, extraOptions)
+  var args = extraOptions.concat(['-sb', '-sn', runCommandOptions]);
```
Comment (simonihmig) [inline lib/utilities/pristine.js]: Not sure why you add the '-sb' and '-sn' options unconditionally although there is `extraOptions` taking care of them?

Comment (kellyselden) [inline lib/utilities/pristine.js]: I saw that npm and bower are installing later anyways:
https://github.com/tomdale/ember-cli-addon-tests/pull/24/files#diff-32914fa89b187f0dba3ec3832f5ad335R109
https://github.com/tomdale/ember-cli-addon-tests/pull/24/files#diff-32914fa89b187f0dba3ec3832f5ad335R123

I figured we could save some cycles by skipping it here.

Comment (simonihmig) [inline lib/utilities/pristine.js]: Seems you are right, indeed. But then we can get rid of the `extraOptions`, as they don't make any difference, right?

Comment (kellyselden) [inline lib/utilities/pristine.js]: Oh I missed the extraOptions setup code above here. I will remove my redundant skips.

**Comment by kellyselden** [comment]
updated per review.

**Comment by simonihmig** [comment]
Seems fine to me! 👍

**Pull request merged**